### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Fixes misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/certificate-transparency-go/actions/runs/6126929463/attempts/1#summary-

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/certificate-transparency-go/actions/runs/6126929623#summary-16631876099
### Checklist

- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -164,7 +165,7 @@ entries whose `start` parameter does not fall on a multiple of the maximum
 permitted number of entries, will have their responses truncated such that
 subsequent requests will align with this boundary.
 This is intended to coerce callers of `get-entries` into all using the same
-`start` and `end` parameters and thereby increase the cachability of
+`start` and `end` parameters and thereby increase the cacheability of
```
Comment (jsoref) [inline CHANGELOG.md:168]: some projects don't like retroactively changing changelogs. Happy to drop this or anything else.

File: submission/distributor_test.go [hunk 2]
```diff
@@ -126,7 +126,7 @@ func sampleUncollectableLogList() *loglist3.LogList {
 	ll.Operators[0].Logs = append(ll.Operators[0].Logs, &loglist3.Log{
 		Description: "Does not return roots", Key: []byte("VW5jb2xsZWN0YWJsZUxvZ0xpc3Q="),
 		URL:   "uncollectable-roots/log/",
-		DNS:   "uncollectavle.ct.googleapis.com",
+		DNS:   "uncollectable.ct.googleapis.com",
```
Comment (jsoref) [inline submission/distributor_test.go:129]: It's possible this was intentional, but I'm hoping it isn't and doesn't matter...

File: submission/races.go [hunk 3]
```diff
@@ -137,7 +137,7 @@ func (sub *safeSubmissionState) setResult(logURL string, sct *ct.SignedCertifica
 		}
 	}
 
-	// Cancel any pending Log-requests for which there're no more awaiting
+	// Cancel any pending Log-requests for which there are no more awaiting
```
Comment (jsoref) [inline submission/races.go:140]: I don't think the abbreviation adds any value. There are on the order of 1000x more uses of the expansion than the contraction according to a Google search...

File: tls/tls.go [hunk 4]
```diff
@@ -638,7 +638,7 @@ func marshalField(out *bytes.Buffer, v reflect.Value, info *fieldInfo) error {
 			}
 		}
 		// Now we have seen all fields in the structure, check that all select(Enum) {..} selector
-		// fields found a source field get get their data from.
+		// fields found a source field to get their data from.
```
Comment (jsoref) [inline tls/tls.go:641]: this is an atypical correction (normally `x x` => `x`)

File: trillian/ctfe/doc.go [hunk 5]
```diff
@@ -17,7 +17,7 @@ Package ctfe contains a usage example by providing an implementation of an RFC69
 log server using a Trillian log server as backend storage via its GRPC API.
 
 IMPORTANT: Only code rooted within this part of the tree should refer to the CT
-Github repository. Other parts of the system must not assume that the data they're
+GitHub repository. Other parts of the system must not assume that the data they're
```
Comment (jsoref) [inline trillian/ctfe/doc.go:20]: brand

File: trillian/ctfe/sth.go [hunk 6]
```diff
@@ -30,7 +30,7 @@ import (
 type contextKey string
 
 // remoteQuotaCtxKey is the key used to attach a Trillian quota user to
-// context.Context passed in to STH getters.
+// context.Context passed into STH getters.
```
Comment (jsoref) [inline trillian/ctfe/sth.go:33]: this correction pattern has been fairly controversial, but I believe this change is correct

File: trillian/integration/hammer.go [hunk 7]
```diff
@@ -499,7 +499,7 @@ func (s *hammerState) addChainInvalid(ctx context.Context) error {
 	return nil
 }
 
-// chooseCertToAdd determines whether to add a new or pre-existing cert.
+// chooseCertToAdd determines whether to add a new or preexisting cert.
```
Comment (AlCutter) [inline trillian/integration/hammer.go:502]: I think [`pre-existing`](https://dictionary.cambridge.org/dictionary/english/pre-existing) is perfectly cromulent :)

File: trillian/migrillian/main.go [hunk 8]
```diff
@@ -51,7 +51,7 @@ var (
 	etcdServers   = flag.String("etcd_servers", "", "A comma-separated list of etcd servers; no etcd registration if empty")
 	lockDir       = flag.String("lock_file_path", "/migrillian/master", "etcd lock file directory path")
 	electionDelay = flag.Duration("election_delay", 0, "Max random pause before participating in master election")
-	backend       = flag.String("backend", "", "GRPC endpoing to connect to Trillian logservers")
+	backend       = flag.String("backend", "", "GRPC endpoint to connect to Trillian logservers")
```
Comment (AlCutter) [inline trillian/migrillian/main.go:54]: I am sad that "endpoing" is not a word - I hereby coin this term to mean an RPC interface which is only ever used to check if a service is alive. :)

File: x509/verify.go
Comment (AlCutter) [inline x509/verify.go:1]: [sic]
This file we'd better leave as-is: it's a fork of the Go `x509` package, with as few modifications as possible (it's also somewhat sketchy to rename exported fields).

**Review by AlCutter — COMMENTED** [review]
Hi @jsoref,

Many thanks for the PR - it's largely good, just a few comments inline.

Cheers,
Al.

**Review by AlCutter — APPROVED** [review]
Thanks for the changes!

Could you possibly rebase onto HEAD? I can merge it then.

**Comment by jsoref** [comment]
@AlCutter: done

**Comment by AlCutter** [comment]
Thanks again, @jsoref!

**Pull request merged**